### ISSUE TITLE
Add a "forbidden" words quality rule

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-server#1025]
Add a "forbidden" words quality rule that will look for insults, vulgar words in ingredients, product name, brands

**Comment by teolemon** [comment]
https://github.com/LDNOOBW/List-of-Dirty-Naughty-Obscene-and-Otherwise-Bad-Words
https://github.com/marteinn/The-Big-Username-Blacklist/blob/master/list_raw.txt

**Comment by CharlesNepote** [comment]
I disagree to build a new quality facet as there will be *many* false positives -- e.g. 7 products use "couille" as part of their name; 1689 products use "bites" in their name; etc.

There should be another process. Example:
* the system alerts some people that XXXX product may contains bad words (in real-time?)
* someone controls the product
* he makes a correction OR leave the product as-is
* he tells the system that the alert has been processed

I think ideally:
* it should be in real-time
* the system should keep the information if a product is a false positive
* it should be replayed

**Comment by teolemon** [comment]
I prefer a quick and dirty facet with false positives to RT notifications with false positives with a feedback loop that probably won't be implemented.

**Comment by CharlesNepote** [comment]
Me either! But why do you think a better solution won't be implemented? There are many developers that could do that.

And we can start with RT notifications without keeping false positive information. If our needs are clearly explained, it shouldn't be so long and difficult.

**Comment by zigouras** [comment]
I can work on this if you guys want.  I think what @teolemon proposed is a good initial plan.  Start with a simple quality facet first and then we can think about adding RT Slack notifications.

**Comment by VaiTon** [comment]
@zigouras you can start working on it! I'm assigning you.

**Comment by teolemon** [comment]
https://world.openfoodfacts.org/product/3168930159896/icetea-saveur-peche-lipton?rev=47

**Comment by teolemon** [comment]
https://fr.openfoodfacts.org/produit/4002566003149/sexe-amour-pipi-caca-bite-cul-gazi?rev=17

**Comment by teolemon** [comment]
Ok, we had a much more problematic issue which stayed undetected for several months.

**Comment by TaciteOFF** [comment]
Found the N word tonight, and obviously not on the product itself

**Comment by CharlesNepote** [comment]
I'm closing because we have built a basic but efficient process to manage textual vandalism. It is published in a private repository: https://github.com/openfoodfacts/private-tools/tree/main/off-vandalism

Don't hesitate to reopen if you feel this issue is still valid.
